src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -114,6 +116,18 @@ internal static FunctionMetadata ReadFunctionMetadata(string scriptDir, ICollect
                 }
                 else if (functionMetadata != null)
                 {
+                    if (functionMetadata.ScriptType == ScriptType.Unknown)
+                    {
+                        var workerConfig = workerConfigs.First(config => config.Extensions.Contains(Path.GetExtension(functionMetadata.ScriptFile)));
```

Comment:
Wouldn't you have to use FirstOrDefault for the behavior you're expecting here? This will throw if you don't have a match.